Keas.Mvc/Services/SecurityService.cs

```diff
@@ -132,43 +137,49 @@ private async Task<bool> IsInRole(Role role, Team team)
         public async Task<User> GetUser()
         {
             var userId = _contextAccessor.HttpContext.User.Identity.Name;
-            var user = await _dbContext.Users.SingleOrDefaultAsync(x => x.Id == userId);
+            var user = await _dbContext.Users.AsNoTracking().SingleOrDefaultAsync(x => x.Id == userId);
             return user;
         }
 
         public async Task<Person> GetPerson(string teamName)
         {
             var userId = _contextAccessor.HttpContext.User.Identity.Name;
             var person =
-                await _dbContext.People.SingleOrDefaultAsync(p => p.User.Id == userId && p.Team.Name == teamName);
+                await _dbContext.People.AsNoTracking().SingleOrDefaultAsync(p => p.User.Id == userId && p.Team.Name == teamName);
             return person;
         }
 
         public async Task<List<User>> GetUsersInRoles(List<Role> roles, int teamId)
         {
-            var users = await _dbContext.TeamPermissions.Where(x => x.TeamId == teamId && roles.Any(r=> r.Id==x.RoleId)).Select(tp=>tp.User).Distinct().ToListAsync();
+            var users = await _dbContext.TeamPermissions.Where(x => x.TeamId == teamId && roles.Any(r=> r.Id==x.RoleId)).Select(tp=>tp.User).Distinct().AsNoTracking().ToListAsync();
             
             return users;
         }
 
         public async Task<List<User>> GetUsersInRoles(List<Role> roles, string teamName)
         {
-            var users = await _dbContext.TeamPermissions.Where(x => x.Team.Name== teamName && roles.Any(r => r.Id == x.RoleId)).Select(tp => tp.User).Distinct().ToListAsync();
+            var users = await _dbContext.TeamPermissions.Where(x => x.Team.Name== teamName && roles.Any(r => r.Id == x.RoleId)).Select(tp => tp.User).Distinct().AsNoTracking().ToListAsync();
 
             return users;
         }
 
         public async Task<List<TeamPermission>> GetUserRolesInTeam(Team team) {
             var userId = _contextAccessor.HttpContext.User.Identity.Name;
-            var userPermissions = await _dbContext.TeamPermissions.Where(x => x.TeamId == team.Id && x.User.Id == userId).ToListAsync();
+            var userPermissions = await _dbContext.TeamPermissions.Where(x => x.TeamId == team.Id && x.User.Id == userId).AsNoTracking().ToListAsync();
+            return userPermissions;
+        }
+
+        public async Task<List<Role>> GetUserRolesInTeamSlug(string teamSlug){
+            var userId = _contextAccessor.HttpContext.User.Identity.Name;
+            var userPermissions = await _dbContext.TeamPermissions.Where(x => x.Team.Slug == teamSlug && x.User.Id == userId).Select(tp=> tp.Role).AsNoTracking().ToListAsync();
```

Comment:
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1061	'Team' does not contain a definition for 'Slug' and no accessible extension method 'Slug' accepting a first argument of type 'Team' could be found (are you missing a using directive or an assembly reference?)	Keas.Mvc	C:\GitProjects\Keas\Keas.Mvc\Services\SecurityService.cs	174	Active

Reply:
Does exist in another branch. But, I ended up not using this in my report controller, so just got rid of it.